### PR TITLE
✨Source Gridly: Migrate Python CDK to Low-code CDK

### LINKED ISSUE #29282: Gridly
https://docs.airbyte.com/integrations/sources/gridly

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
*Migrating Source Gridly to Low-Code CDK*
Closes #29282

## How
*Developed using (Configuration Based Source) low-code CDK*

## Recommended reading order
1. `spec.yaml`
2. `manifest.yaml`
3. `schemas/*`

## Tests

<details><summary><strong>Integration & Acceptance</strong></summary>
Full Test Results

image

</details>

## 🚨 User Impact 🚨
- No breaking changes, migration to low-code
<details><summary><strong>Migrate Connector: Source Gridly</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [x] Connector's `bootstrap.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [x] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

</details>

### CODE REVIEW

**Comment by btkcodedev** [comment]
All tests are passing in local
Checks:
- No pagination support ✅ 
- No incremental support✅ 
- Added parent stream slices✅ 
- Updated version in dockerfile, docs and metadata✅ 

Please ping for the credentials

image

**Comment by marcosmarxm** [comment]
Hello 👋 , right now Airbyte's integration tests are failing for this connector and I'm going to work during the week to fix them. Due a high number of contributions we have a backlog and your contribution will be review next week. We ask to you share the integration test from your side whatever possible and if you don't have credentials to run tests to get in touch with @RealChrisSean as soon as possible. Also some common missing steps contributors does migrating to low code are (and you can double check to make sure you implemented this): 
* adding paginator
* update the documentation changelog + metadata file and dockerfile version
* if the connector has `start_date` parameter you must implement incremental syncs

File: airbyte-integrations/connectors/source-gridly/source_gridly/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      type: "ApiKeyAuthenticator"
+      header: "Authorization"
+      api_token: "ApiKey {{ config['api_key'] }}"
+
+  retriever:
+    type: SimpleRetriever
+    record_selector:
+      $ref: "#/definitions/selector"
+    paginator:
+      type: NoPagination
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-gridly/source_gridly/manifest.yaml:24]: Missing paginator.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by btkcodedev** [comment]
@marcosmarxm But I can't find Pagination support for any stream

**Comment by marcosmarxm** [comment]
> @marcosmarxm But I can't find Pagination support for any stream in previous python source, will try to search documentation

Check https://www.gridly.com/docs/api/#record

**Comment by btkcodedev** [comment]
Thanks @marcosmarxm Pagination added, All tests are passing :)
image

**Comment by sajarin** [comment]
I've been trying to run tests for this PR in another branch but I don't think the implementation of this connector matches the original, it seems to timeout after producing an absurd amount of records (tens of thousands) from the records stream. Unsure if there is an infinite loop somewhere or just an combinatorial explosion of the amount of records read given the tree structure of this api.

**Comment by sajarin** [comment]
Hey @btkcodedev 

After looking at this connector and it's unique functionality, it does not seem possible to convert this to a low-code version. This is because this particular connector dynamically generates the schema which is not supported by the low-code cdk. Correct me if I'm wrong but the current schema files you've created for this connector are only reflective of our sandbox account and will not work in theory for someone else's connection since the schemas would be totally different. 

Unfortunately, we can't accept this PR. Sometimes we get things right but sometimes we make mistake.  I apologize for the time you spent developing this connector, this should have been verified before being listed as a possible contribution. Feel free to reach out if you have any questions.

**Pull request closed without merging**